### PR TITLE
DH-13

### PR DESCRIPTION
**Pull request by Dyrk84**
Boss in the game!
It is possible to call monsters (goblin and goblin king) via the String type parameter.
The monsterCaller method now can call the goblin king too.
If the goblin king is comes, the champion takes damage.
Running away action don't working versus goblin king.
If the goblin king dies, the champion wins.
The champion gets a healing potion at the start.
Added 8 tests.

### CODE REVIEW

File: src/main/java/hu/dungeonhunter/tools/Fight.java [hunk 1]
```diff
@@ -22,11 +27,16 @@
...
     @Setter
     @Getter
-    private int healingPotionCounter = 0;
+    private int randomEnemy;
+
+    @Getter
+    private boolean runningAwayFromGoblinKingTest;
+
+    @Getter
+    private boolean goblinKingIsDeadTest;
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/tools/Fight.java]: You don't need this. You can get this from monster.isDefeat().

File: src/main/java/hu/dungeonhunter/tools/Fight.java [hunk 2]
```diff
@@ -36,72 +46,74 @@ public Fight(int numOfMonsters) {  //for test
...
     public void championAttack() {
         System.out.print("Champion attack: ");
         monster.setHp(monster.getHp() - champion.championDamage());
-        System.out.println("Monster have now " + monster.getHp() + " hit points");
-        monster.enemyVictory();
-        if (!monster.isDefeat()) {
+        System.out.println("The " + monster.getType().charType + " have now " + monster.getHp() + " hit points");
+        if (monster.getType().equals(CharacterTypes.GOBLIN_KING) && monster.getHp() <= 0) {
+            System.out.println("The goblin king is dead! You win!");
+            goblinKingIsDeadTest = true; //toTest
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/tools/Fight.java]: You can test goblin king defeat without this.

File: src/main/java/hu/dungeonhunter/model/CharacterTypes.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package hu.dungeonhunter.model;
+
+import lombok.Getter;
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/model/CharacterTypes.java]: You can remove this import, it is not used.

File: src/main/java/hu/dungeonhunter/tools/Fight.java (hunk 1 above)
Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/tools/Fight.java]: i removed

File: src/main/java/hu/dungeonhunter/tools/Fight.java (hunk 2 above)
Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/tools/Fight.java]: Tests reworked. This boolen removed

File: src/main/java/hu/dungeonhunter/model/CharacterTypes.java (hunk 3 above)
Comment (Dyrk84) [reply src/main/java/hu/dungeonhunter/model/CharacterTypes.java]: Thank you! Removed!

**Review by gergelyszanto — APPROVED** [review]

**Pull request merged**